Tasks/FtpUploadV2/ftpuploadtask.ts

```diff
@@ -23,142 +20,322 @@ export class FtpOptions {
     // advanced options
     clean: boolean;
     cleanContents: boolean;
-    overwrite: boolean; 
-    preservePaths: boolean; 
-    trustSSL: boolean; 
+    preservePaths: boolean;
+    trustSSL: boolean;
+}
+
+class ProgressTracking {
+    ftpOptions: FtpOptions = null;
+    fileCount: number = -1;
+
+    progressFilesUploaded: number = 0;
+    progressDirectoriesProcessed: number = 0;
+
+    constructor(ftpOptions: FtpOptions, fileCount: number) {
+        this.ftpOptions = ftpOptions;
+        this.fileCount = fileCount;
+    }
+
+    directoryProcessed(name: string): void {
+        this.progressDirectoriesProcessed++;
+        this.printProgress(
+            "remote directory successfully created/verified: " + name
+        );
+    }
+
+    fileUploaded(file: string, remoteFile: string): void {
+        this.progressFilesUploaded++;
+        this.printProgress("successfully uploaded: " + file + " to: " + remoteFile);
+    }
+
+    printProgress(message: string): void {
+        let total: number =
```

Comment:
It looks like most of these can be `const` instead of `let`

Reply:
I didn't really bother to change this, this is mostly copied & pasted.  I will take a pass on the variables.